ambry-frontend/src/main/java/com.github.ambry.frontend/AmbrySecurityService.java

```diff
@@ -319,4 +329,39 @@ private void setAccountAndContainerHeaders(RestRequest restRequest, RestResponse
     }
     restResponseChannel.setHeader(RestUtils.Headers.PRIVATE, !container.isCacheable());
   }
+
+  /**
+   * Check precondtions for conditional delete if the {@code restRequest} contains the target
+   * account and container and they are not generic unknowns.
+   * @param restRequest the {@link RestRequest} that contains the {@link Account} and {@link Container} details.
+   * @throws RestServiceException if preconditions check failed.
+   */
+  private void checkForConditionalDelete(RestRequest restRequest) throws RestServiceException {
+    String accountNameFromHeader = getHeader(restRequest.getArgs(), Headers.TARGET_ACCOUNT_NAME, false);
+    String containerNameFromHeader = getHeader(restRequest.getArgs(), Headers.TARGET_CONTAINER_NAME, false);
+    if (accountNameFromHeader != null) {
+      Account targetAccount = (Account) restRequest.getArgs().get(InternalKeys.TARGET_ACCOUNT_KEY);
+      if (targetAccount == null) {
+        throw new RestServiceException("Account cannot be found", RestServiceErrorCode.InvalidAccount);
+      }
+      String accountNameFromBlobId = targetAccount.getName();
+      if (!accountNameFromHeader.equals(accountNameFromBlobId)) {
+        throw new RestServiceException("Account name: " + accountNameFromHeader
+            + " from delete request doesn't match the account name from Blob id : " + accountNameFromBlobId,
+            RestServiceErrorCode.PreconditionFailed);
+      }
+      if (containerNameFromHeader != null) {
+        Container targetContainer = (Container) restRequest.getArgs().get(InternalKeys.TARGET_CONTAINER_KEY);
+        if (targetContainer == null) {
+          throw new RestServiceException("Container cannot be found", RestServiceErrorCode.InvalidContainer);
+        }
+        String containerNameFromBlobId = targetContainer.getName();
+        if (!containerNameFromHeader.equals(containerNameFromBlobId)) {
+          throw new RestServiceException("Container name: " + containerNameFromHeader
```

Comment:
One thing to keep in mind is that this check will only work for blobid v2+. I suppose we can explain that in client-facing docs.